materialize/resources/config.go

```diff
@@ -4,6 +4,9 @@ var connectionTypes = []string{
 	"KAFKA",
 	"POSTGRES",
 	"LOAD GENERATOR",
+	"AWS PRIVATELINK",
```

Comment:
Maybe this should get broken out. Since it's used by `SOURCES` this would make these new options allowable for source configurations.

Reply:
Just changed that, let me know what you think!